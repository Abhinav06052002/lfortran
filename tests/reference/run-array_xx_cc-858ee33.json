{
    "basename": "run-array_xx_cc-858ee33",
    "cmd": "lfortran --continue-compilation --no-color {infile}",
    "infile": "tests/errors/array_xx_cc.f90",
    "infile_hash": "6fb257042311d57514493479dd6cd0ccea2d922d070e572dd398dfae",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "run-array_xx_cc-858ee33.stdout",
    "stdout_hash": "7c30da867d34c3f53085fa5826aac5eebd24829d11e7189553ee7d4a",
    "stderr": "run-array_xx_cc-858ee33.stderr",
    "stderr_hash": "118a37dda92a30d229d7a57a98eee08db8cbb8360560f0eddce4fbf0",
    "returncode": 1
}