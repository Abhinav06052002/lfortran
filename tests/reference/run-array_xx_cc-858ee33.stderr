semantic error: Empty array constructor is not allowed
  --> tests/errors/array_xx_cc.f90:19:5
   |
19 | a = []
   |     ^^ 

semantic error: Empty array constructor is not allowed
  --> tests/errors/array_xx_cc.f90:23:12
   |
23 | print *, [[[], [[]]], [[]], []]
   |            ^^ 

semantic error: Empty array constructor is not allowed
  --> tests/errors/array_xx_cc.f90:24:12
   |
24 | print *, [[[], [[]]], []]
   |            ^^ 

semantic error: Rank mismatch in array reference: the array `b` has rank `1`, but is referenced as rank `2`
  --> tests/errors/array_xx_cc.f90:28:1
   |
28 | b(:,:) = 1
   | ^^^^^^ 

semantic error: Rank mismatch in array reference: the array `b` has rank `1`, but is referenced as rank `2`
  --> tests/errors/array_xx_cc.f90:29:1
   |
29 | b(:,:) = 2
   | ^^^^^^ 

semantic error: Type member x is not an array so it cannot be indexed.
  --> tests/errors/array_xx_cc.f90:33:1
   |
33 | y%x(:) = 1
   | ^^^^^^ 

semantic error: Rank mismatch in array reference: the array `str` has rank `1`, but is referenced as rank `2`
  --> tests/errors/array_xx_cc.f90:37:1
   |
37 | str(1, 2)(:) = '1234'
   | ^^^^^^^^^^^^ 

semantic error: Rank mismatch in array reference: the array `str` has rank `1`, but is referenced as rank `3`
  --> tests/errors/array_xx_cc.f90:38:1
   |
38 | str(1,2,3)(:) = '1234'
   | ^^^^^^^^^^^^^ 

semantic error: Type-spec cannot contain an asterisk for a type parameter
  --> tests/errors/array_xx_cc.f90:42:10
   |
42 | print *, [character(*) :: "a", "b", "ball", "cat"]
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Type-spec cannot contain an asterisk for a type parameter
  --> tests/errors/array_xx_cc.f90:43:10
   |
43 | print *, [character(*) :: "a2", "b2", "ball2", "cat2"]
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Different shape for array assignment on dimension 1(3 and 2)
  --> tests/errors/array_xx_cc.f90:47:1
   |
47 | x1 = reshape([1,2,3,4],[2,2])
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Different shape for array assignment on dimension 1(3 and 1)
  --> tests/errors/array_xx_cc.f90:48:1
   |
48 | x1 = reshape([1,2,3,4],[1,2])
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Arithmetic if (x) requires an integer or real for `x`
  --> tests/errors/array_xx_cc.f90:54:5
   |
54 | if ("yy") 1, 2, 3
   |     ^^^^ 

semantic error: Incorrect number of arguments passed to the 'size' intrinsic. It accepts at least 1 and at most 3 arguments.
  --> tests/errors/array_xx_cc.f90:63:10
   |
63 | size_a = size(a1, 1, 4, kind=4)
   |          ^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Incorrect number of arguments passed to the 'size' intrinsic. It accepts at least 1 and at most 3 arguments.
  --> tests/errors/array_xx_cc.f90:64:10
   |
64 | size_a = size()
   |          ^^^^^^ 

semantic error: Cannot assign to a constant variable
  --> tests/errors/array_xx_cc.f90:68:1
   |
68 | x3 = 1
   | ^^^^^^ assignment here
   |
11 | integer, parameter :: x3 = 2
   |                       ~~~~~~ declared as constant
